l10n_jp_summary_invoice/reports/summary_invoice_reports.xml

```diff
@@ -8,6 +8,10 @@
         <field name="report_file">l10n_jp_summary_invoice.report_summary_invoice</field>
         <field name="binding_model_id" ref="account_billing.model_account_billing" />
         <field name="binding_type">report</field>
+        <field name="attachment_use">True</field>
+        <field
+            name="attachment"
+        >(object.state == 'posted') and ('Summary Invoice-%s' % (object.name))</field>
```

Comment:
```suggestion
        >(object.state == 'billed') and ('Summary Invoice-%s' % (object.name))</field>
```

Please check if assigning date+time attaches the document at every print action. I think the best is to be able to assign a revision number (e.g., _r1, r2...) but I don't have a good idea how to implement this quickly at the moment.